marlowe-playground-client/src/Marlowe/Blockly.purs

```diff
@@ -19,18 +24,15 @@ import Data.Generic.Rep.Enum (genericCardinality, genericFromEnum, genericPred,
 import Data.Generic.Rep.Eq (genericEq)
 import Data.Generic.Rep.Ord (genericCompare)
 import Data.Generic.Rep.Show (genericShow)
-import Data.Maybe (Maybe(..))
-import Data.Traversable (traverse, traverse_)
+import Data.Maybe (Maybe(..), maybe)
+import Data.TraversableWithIndex (forWithIndex)
 import Data.Tuple (Tuple(..))
 import Effect (Effect)
-import Halogen.HTML (HTML)
-import Halogen.HTML.Properties (id_)
-import Marlowe.Holes (Action(..), Bound(..), Case(..), ChoiceId(..), Contract(..), Location(..), Observation(..), Party(..), Payee(..), Term(..), TermWrapper(..), Timeout(..), Token(..), Value(..), ValueId(..), mkDefaultTerm, mkDefaultTermWrapper)
-import Marlowe.Parser as Parser
+import Foreign.Object as Object
+import Marlowe.Holes (Action(..), Bound(..), Case(..), ChoiceId(..), Contract(..), Location(..), Observation(..), Party(..), Payee(..), Term(..), TermWrapper(..), Timeout(..), Token(..), Value(..), ValueId(..))
 import Marlowe.Semantics (Rational(..))
+import Partial.Unsafe (unsafePartial)
```

Comment:
I'm scared

Reply:
embrace the unsafe

![image](https://user-images.githubusercontent.com/2634059/108255459-112b6d80-713b-11eb-8aed-585f0626f91d.png)

Reply:
removed

Reply:
I think the only time you really need anything unsafe is right next to FFI stuff.